src/checkout-buttons/strategies/masterpass/masterpass-button-strategy.ts

```diff
@@ -77,7 +77,13 @@ export default class MasterpassButtonStrategy implements CheckoutButtonStrategy
         const button = document.createElement('input');
 
         button.type = 'image';
-        button.src = `https://${paymentMethod.config.testMode ? 'sandbox.' : ''}src.mastercard.com/assets/img/btn/src_chk_btn_126x030px.svg?locale=${storeConfig.storeProfile.storeLanguage}&paymentmethod=master,visa,amex,discover&checkoutid=${paymentMethod.initializationData.checkoutId}`;
+
+        if (paymentMethod.initializationData.isMasterpassSrcEnabled) {
+            button.src = `https://${paymentMethod.config.testMode ? 'sandbox.' : ''}src.mastercard.com/assets/img/btn/src_chk_btn_126x030px.svg?locale=${storeConfig.storeProfile.storeLanguage}&paymentmethod=master,visa,amex,discover&checkoutid=${paymentMethod.initializationData.checkoutId}`;
```

Comment:
this is quite hard to read/follow, I would recommend splitting it into different lines, e.g.
```js
const subdomain = paymentMethod.config.testMode ? 'sandbox.' : '';
const { storeLanguage } = storeConfig.storeProfile;
const { checkoutId } = paymentMethod.initializationData;

const params = [
    `locale=${storeLanguage}`,
    `paymentmethod=master,visa,amex,discover`,
    `checkoutid=${paymentMethod.initializationData}`
]

button.src = [
    `https://${subdomain}src.mastercard.com/assets/img/btn/src_chk_btn_126x030px.svg',
    params.join('&')
].join('?');